### PR TITLE
Update Jetty Maven Plugin to 9.4.12.v20180830

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This change aligns Jetty Maven Plugin codebase with version 9.4.12.v20180830 currently used in Jenkins. Works on my machine (c), no Jenkinsfile in this branch 

https://issues.jenkins-ci.org/browse/JENKINS-54599

@jenkinsci/java11-support

### CODE REVIEW

**Comment by batmat** [comment]
@oleg-nenashev Sorry, I'm missing something. I see 127 files modified. Possibly intended, but I need to better understand the scope of this, which for now does not match my expectation of just bumping Jetty version :-). Looking more deeply.

File: Jenkinsfile
Comment (batmat) [inline Jenkinsfile:25]: `verify` is generally better than `install`, but I suppose it's fine here.

File: Jenkinsfile
Comment (batmat) [inline Jenkinsfile]: `allowEmptyResults: true` are you really sure you want this? I think it's not great, given there *are* tests currently and we'd rather catch it if for some reason no more tests are detected at the end.

Or at least, please explain why it's irrelevant here.

**Comment by oleg-nenashev** [comment]
@batmat See the repository README. 

"Maven plugin for developing Jenkins core This Maven plugin is intended to be the single plugin to contain all the goals needed for hacking core. IOW, the core equivalent of maven-hpi-plugin. In reality, this is a **patched version of jetty-maven-plugin** that defines only one goal hudson-dev:run to run the web application. "

So I follow the process in https://github.com/jenkinsci/maven-hudson-dev-plugin#upstream-tracking

**Review by batmat — CHANGES_REQUESTED** [review]
I would prefer we do not let missing tests go possibly unnoticed, the rest LGTM

File: Jenkinsfile
Comment (oleg-nenashev) [reply Jenkinsfile]: Removed the flag. Jenkinsfile was just copy-pasted from the master branch where there were no tests

**Review by batmat — APPROVED** [review]
LGTM, provided PR build still passes, obviously.

**Pull request merged**
